### DISCUSSION TITLE
dem.py failing

### DISCUSSION THREAD

**Discussion by feigl** [isce-framework/isce2#548] [General]
Hi All,

I am struggling with dem.py. It fails for several attempts as detailed below

- 1 arc sec
- 3 arc sec
- curl on command line
- browser while logged into earthdata

I know that the password in my $HOME/.netrc file is valid.

Is the issue related to cookies? Or is the URL invalid?

Thanks,

Kurt

### Try with 1 arcsec
(verdi) ops@ebe8970534ad:/Users/feigl/testdir$ dem.py -a stitch -b 38 39 -112 -111 -r -s 1 -c
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed sofware at
https://winsar.unavco.org/software/isce
curl -n  -L -c $HOME/.earthdatacookie -b $HOME/.earthdatacookie -k -f -O http://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL1.003/2000.02.11/N38W112.SRTMGL1.hgt.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (7) Failed to connect to e4ftl01.cr.usgs.gov port 80: Connection refused
2022-08-04 21:25:22,811 - isce.contrib.demUtils.DemStitcher - WARNING - There was a problem in retrieving the file  http://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL1.003/2000.02.11/N38W112.SRTMGL1.hgt.zip. Exception 
Could not create a stitched DEM. Some tiles are missing
N38W112.SRTMGL1.hgt.zip = failed

### try with 3 arc sec
(verdi) ops@ebe8970534ad:/Users/feigl/testdir$ dem.py -a stitch -b 38 39 -112 -111 -r -s 3 -c
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed sofware at
https://winsar.unavco.org/software/isce
curl -n  -L -c $HOME/.earthdatacookie -b $HOME/.earthdatacookie -k -f -O http://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL3.003/2000.02.11/N38W112.SRTMGL3.hgt.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (7) Failed to connect to e4ftl01.cr.usgs.gov port 80: Connection refused
2022-08-04 21:45:11,365 - isce.contrib.demUtils.DemStitcher - WARNING - There was a problem in retrieving the file  http://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL3.003/2000.02.11/N38W112.SRTMGL3.hgt.zip. Exception 
Could not create a stitched DEM. Some tiles are missing
N38W112.SRTMGL3.hgt.zip = failed

### Try with just a curl command
(base) brady:~ feigl$ curl -n  -L -c $HOME/.earthdatacookie -b $HOME/.earthdatacookie -k -f -O http://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL1.003/2000.02.11/N38W112.SRTMGL1.hgt.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (7) Failed to connect to e4ftl01.cr.usgs.gov port 80 after 60 ms: Connection refused

### Try with just a browser (chrome), when I am logged into Earthdata
This site can’t be reached
e4ftl01.cr.usgs.gov refused to connect.

**Comment by fdelgadodelapuente** [comment]
Sometimes the USGS server goes into maintenance and you can't download SRTM in the meantime. That's why I it's a good idea to keep a copy of SRTM somewhere

**Reply by sisi-ali** [reply]
hi every one.
I can not download Dem. 
What is the solution to download and use ISCE software.

![18ea23c6-aff8-4a23-a556-0cc8537178c8](https://user-images.githubusercontent.com/108397906/183157170-e5fbb40a-b13a-4649-880a-ddde9e246f94.jpg)
?

**Reply by EJFielding** [reply]
Hi Kurt,

I tracked down the problem in the middle of August. The LP DAAC changed their data servers to require `https:` instead of `http:`.  This meant that we had to change the ISCE code in a number of places where the SRTM URL was specified. The changes have been merged into the main version of ISCE now.

**Reply by Carolinah23** [reply]
Hello Dr. Fielding,

I had a similar issue to Kurt, then I reinstalled ISCE2 (in a new conda environment), and downloaded the DEM but I still got an error, do you think there is an issue with my installation?  
![image](https://user-images.githubusercontent.com/54252458/190512848-7812f580-4030-4fb9-a4f6-152f5ab76b91.png)
These are the files that I could download (I can display the file .dem.vrt in qgis and looks fine).
![image](https://user-images.githubusercontent.com/54252458/190513104-7ed8506d-4206-4c45-87b6-73c253fd54d1.png)

Thanks!

EDIT: I am using topsApp.py

**Reply by EJFielding** [reply]
Hello Carolina,

We had some trouble with getting the latest version of ISCE2 (v2.6.1) to compile on conda-forge, so if you use `conda install isce2` you will only get the v2.6.0 version that does not include the DEM URL fixes until we resolve the problem.

It is possible to go into the isce2 installation in your conda environment and replace the 5-6 Python programs that include the SRTM URL. Eric Lindsey described a script to do that change in another thread.

++Eric

**Comment by aditdp** [comment]
This issue because of the lpdaac server still in maintenance as you can see the news in this link 
(https://lpdaac.usgs.gov/news/lp-daac-data-access/)


As the alternative, I used the -u flag in the command and retrieve the DEM from ESA DEM SRTM repository in this link (http://step.esa.int/auxdata/dem/SRTMGL1/)

for example:

`dem.py -a stitch -b 38 39 -112 -111 -r -s 1 -c -u http://step.esa.int/auxdata/dem/SRTMGL1/`

Hope this alternative worked, for the dem retrieval issue

**Reply by feigl** [reply]
Thanks! Works for me.

**Reply by nadiahassanpour** [reply]
> This issue because of the lpdaac server still in maintenance as you can see the news in this link (https://lpdaac.usgs.gov/news/lp-daac-data-access/)
> 
> As the alternative, I used the -u flag in the command and retrieve the DEM from ESA DEM SRTM repository in this link (http://step.esa.int/auxdata/dem/SRTMGL1/)
> 
> for example:
> 
> `dem.py -a stitch -b 38 39 -112 -111 -r -s 1 -c -u http://step.esa.int/auxdata/dem/SRTMGL1/`
> 
> Hope this alternative worked, for the dem retrieval issue

Hi. i have this problem now and i can not download dem.py with this script that you say and it failed. could you tell me how can i solve it?

**Reply by EJFielding** [reply]
The latest version of ISCE2 on GitHub has the fixes to use `https://` instead of `http://` for the LP DAAC servers of SRTM data. If you install from the GitHub source, you will get this version that works. If you install from `conda-forge`, it presently has version 2.6.1 that does not have the fix, but you can fix the Python code in your installation.

**Reply by EJFielding** [reply]
See this post for a quick way to patch the ISCE2 code. I have used this on Conda installations.
https://github.com/isce-framework/isce2/issues/549#issuecomment-1250204727

**Reply by Leijunyang** [reply]
> This issue because of the lpdaac server still in maintenance as you can see the news in this link (https://lpdaac.usgs.gov/news/lp-daac-data-access/)
> 
> As the alternative, I used the -u flag in the command and retrieve the DEM from ESA DEM SRTM repository in this link (http://step.esa.int/auxdata/dem/SRTMGL1/)
> 
> for example:
> 
> `dem.py -a stitch -b 38 39 -112 -111 -r -s 1 -c -u http://step.esa.int/auxdata/dem/SRTMGL1/`
> 
> Hope this alternative worked, for the dem retrieval issue

Hello, I solved part of the problem by using this method, but I visited and found that http://step.esa.int/auxdata/dem/SRTMGL1/ does not have DEM data above 60 degrees, what can I do if I want to download this data.
for example 
dem.py -a stitch -b 61 62 -112 -111 -r -s 1 -c -u http://step.esa.int/auxdata/dem/SRTMGL1/ 
It can not download DEM.

**Reply by EJFielding** [reply]
Hello,
The SRTM data was not acquired north of 60 N or south of 60 S due to the orbit of the NASA Space Shuttle. You should look at using the Copernicus DEM available from the Copernicus servers that has full global coverage.

**Comment by sisi-ali** [comment]
Thanks Dear aditdp

**Comment by sisi-ali** [comment]
hi Dear aditdp
You know the accuracy of the DEM downloaded from this site for which range is the highest?

**Reply by aditdp** [reply]
It's actually the same 1 arc second resolution dem

**Comment by sisi-ali** [comment]
Hi
Prasetio  I want to calculate the closure_phase_bias.py using the MintPy
software that I per-processed on ISCE,
but I cant calculate with (
https://github.com/insarlab/MintPy/blob/main/mintpy/closure_phase_bias.py)
I would be grateful if you could guide me

‫‪Prasetio‬‏ ***@***.***‬‏> در تاریخ پنجشنبه ۱۱ اوت ۲۰۲۲
ساعت ۱۴:۲۷ نوشت:‬
